docs/src/main/paradox/define-actors.md

```diff
@@ -1,44 +1,77 @@
 ## Defining Actors and messages      
 
-Messages can be of arbitrary type (any subtype of `Object`). You can send boxed primitive values (such as `String`, `Integer`, `Boolean` etc.) as messages as well as plain data structures like arrays and collection types. 
+Each actor defines a type `T` for the messages it can receive.
+Case classes and case objects make excellent messages since they are immutable and have support for pattern matching, 
+something we will take advantage of in the Actor when matching on the messages it has received. 
 
-Hello World Actors use three different messages:
+The Hello World Actors use three different messages:
 
-* `WhoToGreet`: recipient of the greeting
-* `Greet`: instruction to execute greeting
-* `Greeting`: message containing the greeting
+* `Greet`: command sent to the `Greeter` actor to greet
+* `Greeted`: reply from the `Greeter` actor to confirm the greeting has happened
+* `Start`: command to the `GreeterMain` to start the greeting process
 
 When defining Actors and their messages, keep these recommendations in mind:
 
-* Since messages are the Actor's public API, _it is a good practice to define messages with good names and rich semantic and domain specific meaning_, even if they just wrap your data type. This will make it easier to use, understand and debug actor-based systems.
+* Since messages are the Actor's public API, _it is a good practice to define messages with good names and rich semantic and domain specific meaning_, even if they just wrap your data type. This will make it easier to use, understand and debug actor-based systems. 
 
-* _Messages should be immutable_, since they are shared between different threads. 
+* _Messages should be immutable_, since they are shared between different threads.
 
-* _It is a good practice to put an actor's associated messages as static classes in the class of the Actor_. This makes it easier to understand what type of messages the actor expects and handles. 
+* It is a good practice to put an actor's associated messages in its object. This makes it easier to understand what type of messages the actor expects and handles. 
```

Comment:
its object is from Scala, still something like "as static classes in the class of the Actor"